src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -427,6 +426,10 @@ void start() {
                 if (adPipe != null && adPipe.available() > 0) {
                     byte[] buffer = new byte[0];
                     boolean terminated = false;
+                    ByteArraySearcher configTermSearcher
+                            = new ByteArraySearcher(App.AD_CONFIG_TERM.getBytes());
+                    ByteArraySearcher legacyConfigTermSearcher
+                            = new ByteArraySearcher(App.AD_LEGACY_CONFIG_TERM.getBytes());
```

Comment:
these searchers are stateless and are derived from constant strings, so they could be made static (or at least created outside of the while loop) and re-used  on each iteration

Reply:
Yes, but I don't think this loop has been written with efficiency in mind and this would move their declarations a long way from their use.